pydoctor/extensions/zopeinterface.py

```diff
@@ -297,18 +310,18 @@ def visit_Call_zope_interface_classImplements(self, funcName: str, node: ast.Cal
                                 funcName == 'zope.interface.classImplementsOnly')
     visit_Call_zope_interface_classImplementsOnly = visit_Call_zope_interface_classImplements
 
-    def visit_ClassDef(self, node: ast.ClassDef) -> Optional[ZopeInterfaceClass]:
-        cls = super().visit_ClassDef(node)
-        if cls is None:
-            return None
-        assert isinstance(cls, ZopeInterfaceClass)
+    def depart_ClassDef(self, node: ast.ClassDef) -> None:
+        cls = self.visitor.builder.current.contents.get(node.name)
```

Comment:
Would be good to edit the builder to have something like this instead:

```suggestion
        cls = self.visitor.builder.last
```

Reply:
I'll merge this PR as is create a follow-up PR to refactor this, it might more complicated than expected.